### ISSUE TITLE
`argo submit --serviceaccount` doesn't seem to have an effect

### ISSUE THREAD

**Issue by cujeff09** [argoproj/argo#3954]
**Summary**
We are upgrading from Argo 2.6.1 to 2.9.5.  In the past, we've always provided admin privileges to the default service account, and we've had no issues.  In 2.9.5, the same is true.  However, we wanted to make our execution "more correct" but not using the admin and instead forcing our argo submits to use the argo serviceaccount generated by the installation.

Our setup is like this:  We've installed to the argo namespace the 2.9.5 tag of the install.yaml, applied our workflowtemplates, and started a custom pod that receives messages from an external source, extracts information, and runs `argo submit`.  Without passing in `--serviceaccount`, the received error is as expected per https://argoproj.github.io/argo/service-accounts/:
```
Failed to submit workflow: templates.common-workflow-steps.steps[0].read-properties workflowtemplates.argoproj.io "read-properties-template" is forbidden: User "system:serviceaccount:argo:default" cannot get resource "workflowtemplates" in API group "argoproj.io" in the namespace "argo"
```
When we pass in `--serviceaccount`, however, the error message is unchanged; it still references system:serviceaccount:argo:default.  Our command is this this:
```
argo submit --namespace argo --serviceaccount argo /tmp/workflow/common-workflow.yaml --generate-name about-build-12345 -p git-url=git@hidden.git -p project=about -p commit=12345
```
Not knowing the correct format for the string argument to `--serviceaccount`, we've tried argo, argo:argo, and system:serviceaccount:argo:argo.

If we add admin privileges to default, the workflow executes successfully.

**What is Expected**
When the `--serviceaccount argo` flag is provided, the selected service account should be argo (installed with the 2.9.5 installation) and not default.

**Comment by alexec** [comment]
Interesting. When does the error appear? When you try to submit, or in the logs of the workflow?

**Comment by cujeff09** [comment]
We figured out the underlying issue.  I mentioned that we have a custom pod running that receives data and calls `argo submit`.  It runs with the default service account.  Since it spawns `argo submit` with the argo CLI internally, Kubernetes sets the service account of the initial pod creation as default.  Then, all subsequent pods as created by the workflow-controller are set with the argo service account as specified with `--serviceaccount argo`.  It's the initial creation that causes the call to fail.

We tested a temporary fix by deploying our custom pod with the argo service account and dropping default admin privileges and then verifying the service accounts of all pods created throughout the workflow.  Our final fix is to create a custom service account with the necessary argoproj.io permissions that we'll run the custom pod from and then use your generated argo service account for all pods as part of the workflow.

I'm closing this issue.  Thanks for your help.
